src/controller/CHIPDeviceController.cpp

```diff
@@ -894,8 +894,34 @@ CHIP_ERROR DeviceCommissioner::StopPairing(NodeId remoteDeviceId)
 
 CHIP_ERROR DeviceCommissioner::UnpairDevice(NodeId remoteDeviceId)
 {
-    // TODO: Send unpairing message to the remote device.
-    return CHIP_NO_ERROR;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    MATTER_TRACE_EVENT_SCOPE("UnpairDevice", "DeviceCommissioner");
+
+    Platform::UniquePtr<DeviceUnpair> deviceUnpair(new DeviceUnpair());
+    deviceUnpair->RegisterCallback(this);
+    CommissioneeDeviceProxy * commissioneeDeviceProxy = nullptr;
+    if (CHIP_NO_ERROR == GetDeviceBeingCommissioned(remoteDeviceId, &commissioneeDeviceProxy))
+    {
+        deviceUnpair->UnpairDevice(commissioneeDeviceProxy, remoteDeviceId);
```

Comment:
What does it mean to RemoveFabric for a PASE session?  I guess after AddNOC it'll do something useful, but just expiring the failsafe would do exactly the same thing, right?

More to the point, what do the callers of this API expect to happen for the PASE case?

Reply:
I don't know if I'm misunderstanding Matter Spec. However, I thought UnpairDevice meant to disconnect the controller and the controller that had already been commissioned. Therefore, we used removeFabric. Also, this implementation was carried out by referring to the code below. (only darwin platform implemented)
https://github.com/project-chip/connectedhomeip/blob/fb185956b987859633acc008bcba17db64d0e3df/src/darwin/CHIPTool/CHIPTool/Framework%20Helpers/DefaultsUtils.m#L173

Reply:
I don't know if I'm misunderstanding Matter Spec. However, I thought UnpairDevice meant to disconnect the controller and the controller that had already been commissioned. Therefore, we used removeFabric. Also, this implementation was carried out by referring to the code below. (only darwin platform implemented)
https://github.com/project-chip/connectedhomeip/blob/fb185956b987859633acc008bcba17db64d0e3df/src/darwin/CHIPTool/CHIPTool/Framework%20Helpers/DefaultsUtils.m#L173

Reply:
> I don't know if I'm misunderstanding Matter Spec. However, I thought UnpairDevice meant

There is no concept of "unpair" in the spec.

The API means ... whatever it means.  `MTRUnpairDeviceWithID` is part of an example application, not part of the Matter.framework, precisely because what this operation should do depends on the application.

Note that `MTRUnpairDeviceWithID` will only work with CASE sessions and will not try to do anything if there is no operational device with the given node id, by the way.

Reply:
I think the purpose of this API is to remove Fabric for deviceUnpair. Of course, I know that this process is meaningful after AddNOC, but I think it will only be used after PASE when using this API.